pkg/services/govmomi/vcenter/clone_test.go

```diff
@@ -16,15 +16,14 @@ limitations under the License.
 
 package vcenter
 
-//nolint:all
+//nolint:goimports
```

Comment:
This can be fixed by running `make lint-fix` so we shouldn't need to ignore here.

Reply:
ack, please drop goimports everywhere and run `make lint-fix` afterwards 👍

Reply:
Hi, folks, I tried to drop `goimports` and use `make lint-fix` to fix this. But the diff with the previous one is it just adds a blank line between the duplicated package name. And when I re-run the `golangci-lint run`, it comes up with another error message, saying `File is not `gci`-ed with --skip-generated -s standard -s default (gci)`. Seem like those two linter attempt to break each other, do we have any suggestions on this?

```
import (
	ctx "context"
	"crypto/tls"
	"testing"

	"github.com/vmware/govmomi/object"
	"github.com/vmware/govmomi/simulator"

	// run init func to register the tagging API endpoints.
	_ "github.com/vmware/govmomi/vapi/simulator"
	"github.com/vmware/govmomi/vim25/types"

	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
)
```

Reply:
Should we 
- Add an `importas alias` for `github.com/vmware/govmomi/simulator` (Since this file is only for test purposes, and maybe it's not necessary to change the `.golangci.yml` file for it)
- Add `//nolint: gci` to ignore the new error.
- Add `//nolint: goimports`. (current solution)
or guys have any other suggestions?

Reply:
What happens if you drop the "blank" line from the config?

Reply:
> What happens if you drop the "blank" line from the config?

Then everything goes back to its original way, which is exactly the same when `local-prefixes:` is enabled.

Reply:
Actually, if we don't change the `.golangci.yml` file, in other words, when `local-prefixes` is enabled, then running `golangci-lint run --fix` multiple times will result in the code alternating between having a blank line under the `github.com/vmware/govmomi/simulator` package and not having one.

Reply:
I think you need to add `custom-order` too so it looks like:
```
  gci:
    sections:
      - standard # Standard section: captures all standard packages.
      - default # Default section: contains all imports that could not be matched to another section type.
      - prefix(sigs.k8s.io/cluster-api-provider-vsphere) # Custom section: groups all imports with the specified Prefix.
    custom-order: true
```

see the [docs](https://golangci-lint.run/usage/linters/#gci)

Reply:
> I think you need to add `custom-order` too so it looks like:
> 
> ```
>   gci:
>     sections:
>       - standard # Standard section: captures all standard packages.
>       - default # Default section: contains all imports that could not be matched to another section type.
>       - prefix(sigs.k8s.io/cluster-api-provider-vsphere) # Custom section: groups all imports with the specified Prefix.
>     custom-order: true
> ```
> 
> From the docs:
> 
> > If `custom-order` is `true`, it follows the order of `sections` option.
> 
> see the [docs](https://golangci-lint.run/usage/linters/#gci)

Sure! (BTW, the default order of `gci` section is standard > default > custom > blank > dot, and enabling `custom-order` will make it more readable and robust.)

Reply:
Note: I opened https://github.com/kubernetes-sigs/cluster-api/pull/9339 to do the same there.

Reply:
So folks, are we good to enable `sections:blank`? 😄

Reply:
I wouldn't add a separate section for blanks for now. Just because we have to make a case for it in core CAPI and I'm not sure if that is worth the effort

Reply:
> I wouldn't add a separate section for blanks for now. Just because we have to make a case for it in core CAPI and I'm not sure if that is worth the effort to make the case (and if it's actually worth it introducing it)

Make sense to me, I'll find an alternate solution.

Reply:
Just come up with an idea, Maybe there is a bug of `goimports` when the imported package has a comment above. And I tested with the below format, it works fine. Without introducing the `sections:blank`.
```
import (
	ctx "context"
	"crypto/tls"
	"testing"

	"github.com/vmware/govmomi/object"
	"github.com/vmware/govmomi/simulator"
	_ "github.com/vmware/govmomi/vapi/simulator" // run init func to register the tagging API endpoints.
	"github.com/vmware/govmomi/vim25/types"

	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
)
```
Does this solution seem like a good option?

Reply:
That works. Just out of curiosity. If you format the block like this:
```go
import (
	ctx "context"
	"crypto/tls"
	"testing"
	"github.com/vmware/govmomi/object"
	"github.com/vmware/govmomi/simulator"
	// run init func to register the tagging API endpoints.
	_ "github.com/vmware/govmomi/vapi/simulator"
	"github.com/vmware/govmomi/vim25/types"
	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
)
```
And then run `make lint-fix`. It doesn't come up with a valid formatting? (without further findings)

Reply:
if I run `make lint-fix`, the format will become like this:
```
import (
	ctx "context"
	"crypto/tls"
	"testing"

	"github.com/vmware/govmomi/object"
	"github.com/vmware/govmomi/simulator"

	// run init func to register the tagging API endpoints.
	_ "github.com/vmware/govmomi/vapi/simulator"
	"github.com/vmware/govmomi/vim25/types"

	infrav1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/context"
	"sigs.k8s.io/cluster-api-provider-vsphere/pkg/session"
)
```
And when you run `golangci-lint run` command with this format, it will come up with another error, says: "pkg/services/govmomi/vcenter/clone_test.go:26: File is not `gci`-ed with --skip-generated -s standard -s default -s prefix(sigs.k8s.io/cluster-api-provider-vsphere) --custom-order (gci)"

Reply:
I think this is because of the comment. Maybe we can inline the comment

```go
	_ "github.com/vmware/govmomi/vapi/simulator" 	// run init func to register the tagging API endpoints.
```

Reply:
Got it, thx for trying it. Yup let's go with your suggestion (https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/2294#discussion_r1309956345)